staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -199,6 +203,14 @@ func (s *SpdyRoundTripper) dialWithHttpProxy(req *http.Request, proxyURL *url.UR
 	return rwc, nil
 }
 
+type fnDialer struct {
```

Comment:
is this only needed because `proxy.SOCKS5` has a Dialer interface as argument?

Reply:
fwiw, proxy.SOCKS5 uses a context dialer if passed in, and returns an implementation that implements context dialing

https://github.com/kubernetes/kubernetes/blob/c56629803d4a34377110499a6e9985bd7d5b9d0c/vendor/golang.org/x/net/proxy/socks5.go#L17-L23

https://github.com/kubernetes/kubernetes/blob/c56629803d4a34377110499a6e9985bd7d5b9d0c/vendor/golang.org/x/net/internal/socks/socks.go#L151

Does this wrapping let the socks proxy dialer successfully recognize context dialing is supported here?

Reply:
Good catch! fixing.